### ISSUE TITLE
Improve tree positioning in visualization

### ISSUE THREAD

**Issue by iSaran** [miccol/ROS-Behavior-Tree#12]
Hi again @miccol 

I noticed a problem in the visualization of the tree in which some nodes are visualized on top of each other making harder to observe the tree. If this is the case and is not a problem of my set-up only, I intend to implement in my free time the Reingold-Tilford algorithm for tree nodes positioning for the behavior trees as described in the following papers.

```
1. Buchheim, C., & Michael, J. (2002). Improving Walker ’ s Algorithm to Run in Linear Time, 344–353.
2. Ii, J. Q. W., & Hall, S. (1989). A Node-Positioning Algorithm for General Trees, 20(December 1988), 685–705.
```

Let me know if this problem exist (and is not only me) and if you are interested to start work on that.

Iason

**Comment by miccol** [comment]
I am sure I replied to this over email but I cannot see it here.

Press:  x + PagUp/PagDown to increase/decrease the spacing between the nodes at depth x.

It is a quick and dirty fix but I had problems in estimating the node size (it depends on the label length) using GLUT.

Of course, if you find a smarter way to do that (I am sure there is), feel free to add it.

Michele

**Comment by iSaran** [comment]
Instead of implementing an algorithm for the existent visualization I decided to use the DOT language, which I see is used widely for the visualization of graphs with an aesthetic pleasing way.

Finally, it will be possible to visualize the tree in RQT, by publishing DOT code in ROS. As this is a ROS package, I do not see an issue for the dependency in rqt. In case someone want to use the library without ROS , they can always use your GLUT visualization.

I am working on this branch https://github.com/miccol/ROS-Behavior-Tree/compare/master...iSaran:add_rqt_dot_visualization

I will open a PR as soon as I tested and fully documented.
